### PR TITLE
ts port - removed unused dependencies

### PR DESCRIPTION
**Pull request by PopGoesTheWza**
I wanted to check if a .d.ts could be done with this library. Found some vulnerable and unused dependencies, then wanted to give a look at how ava could work with typescript... got sidetracked and here it is.

`npm install && npm run build && npm run lint && npm run test`

At least you may just grab the .d.ts and update the package.json ;)

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
+"use strict";
```
Comment (grant) [inline lib/index.js]: This file needs to be ignored.

File: src/index.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
+import { chmodSync, constants, exists, unlinkSync } from 'fs';
+import { homedir } from 'os';
+import { join } from 'path';
+import { readFile, writeFile } from 'jsonfile';
+
+// tslint:disable-next-line:no-any
+type jsonAny = any;
```
Comment (grant) [inline src/index.ts]: I think we can just use `object` instead of this type.
Delete this.

Comment (PopGoesTheWza) [reply src/index.ts]: With unknown json object, `any` is usually the best option... Unless I can add generics into the definition.

File: lib/index.js (hunk 1 above)
Comment (PopGoesTheWza) [reply lib/index.js]: Please clarify/elaborate. I am unsure what you mean. Change tsconfig.json so that no `use strict` is emited?

Comment (PopGoesTheWza) [reply lib/index.js]: Oh... You probaly mean that *.js gets added to .gitignore, correct?

**Comment by PopGoesTheWza** [comment]
No need to rush, but once published, Clasp should be able to properly import it with its .d.ts

**Pull request merged**